provider/pkg/autonaming/application.go

```diff
@@ -41,29 +59,57 @@ func ApplyAutoNaming(
 // based on its URN name, It ensures the name meets the length constraints, if known.
 // Defaults to the name followed by 7 random hex characters separated by a '-'.
 func getDefaultName(
-	randomSeed []byte,
 	urn resource.URN,
+	engineAutonaming EngineAutonamingConfiguration,
 	autoNamingSpec *metadata.AutoNamingSpec,
 	olds,
 	news resource.PropertyMap,
 	config *AutoNamingConfig,
-) (resource.PropertyValue, error) {
+) (*resource.PropertyValue, bool, error) {
```

Comment:
Nit: Is the boolean just used to indicate whether a name is returned? I feel like we could omit that because we should non-nil/non-empty check the name to ensure a bug in autonaming is not silently setting a name to the empty string. (i.e. if no error is returned but the name is empty we know something is fishy)

Just a nit, feel free to ignore

Reply:
I'm probably overthinking it indeed. Pushed a commit to simplify back to two return values and nil checks.